File: clients/python/debugging/color.py
from stream_wrapper import StreamWrapper

class Color:
    """RGBA Color"""

    __slots__ = ("r","g","b","a",)

    r: float
    g: float
    b: float
    a: float

    def __init__(self, r: float, g: float, b: float, a: float):
        self.r = r
        """Red component"""
        self.g = g
        """Green component"""
        self.b = b
        """Blue component"""
        self.a = a
        """Alpha (opacity) component"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Color":
        """Read Color from input stream
        """
        r = stream.read_double()
        g = stream.read_double()
        b = stream.read_double()
        a = stream.read_double()
        return Color(r, g, b, a)
    
    def write_to(self, stream: StreamWrapper):
        """Write Color to output stream
        """
        stream.write_double(self.r)
        stream.write_double(self.g)
        stream.write_double(self.b)
        stream.write_double(self.a)
    
    def __repr__(self):
        return "Color(" + \
            repr(self.r) + \
            ", " + \
            repr(self.g) + \
            ", " + \
            repr(self.b) + \
            ", " + \
            repr(self.a) + \
            ")"

File: clients/javascript/debugging/color.js
/**
 * RGBA Color
 */
class Color {
    /**
     * Red component
     */
    r;
    /**
     * Green component
     */
    g;
    /**
     * Blue component
     */
    b;
    /**
     * Alpha (opacity) component
     */
    a;

    constructor(r, g, b, a) {
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }

    /**
     * Read Color from input stream
     */
    static async readFrom(stream) {
        let r;
        r = await stream.readDouble();
        let g;
        g = await stream.readDouble();
        let b;
        b = await stream.readDouble();
        let a;
        a = await stream.readDouble();
        return new Color(r, g, b, a);
    }

    /**
     * Write Color to output stream
     */
    async writeTo(stream) {
        let r = this.r;
        await stream.writeDouble(r);
        let g = this.g;
        await stream.writeDouble(g);
        let b = this.b;
        await stream.writeDouble(b);
        let a = this.a;
        await stream.writeDouble(a);
    }
}
module.exports = Color